src/pretix/presale/views/cart.py

```diff
@@ -630,8 +630,12 @@ def dispatch(self, request, *args, **kwargs):
         self.subevent = None
         if request.event.has_subevents:
             if request.GET.get('subevent'):
-                self.subevent = get_object_or_404(SubEvent, event=request.event, pk=request.GET.get('subevent'),
-                                                  active=True)
+                try:
+                    subevent_pk = int(request.GET.get('subevent'))
+                except ValueError:
+                    raise Http404(_('No SubEvent matches the given query.'))
```

Comment:
We should never ever use the word "SubEvent" when talking to users, it should also never show up in translations. 

I know you copied this from the auto-generated message by get_object_or_404 as well, but maybe we should change both then?

Correct would be e.g.:

```
raise Http404(pgettext('subevent', 'We were unable to find the specified date.'))
```

(We always use pgettext for subevent because "date" is ambigous in english between a "subevent" and a "calendar date", but we use different words eg in German (Termin vs Datum))

Reply:
Thanks very much, your message makes way more sense – and yes, I copied it from `get_object_or_404` :) I will change it for both of them.